api/app/controllers/spree/api/payments_controller.rb

```diff
@@ -17,8 +17,10 @@ def new
       end
 
       def create
-        @payment = @order.payments.build(payment_params)
-        if @payment.save
+        success = @order.contents.update_cart(payments_attributes: [payment_params])
+        @payment = @order.payments.last
+
+        if success
           respond_with(@payment, status: 201, default_template: :show)
         else
           invalid_resource!(@payment)
```

Comment:
would this respond with the correct payment in a failed scenario? it wouldn't be the order's last payment would it?